#!/usr/bin/env python

import traceback
import sys
import os

def excuse():
    except_message = traceback.format_exc()
    stack_message  = string.join(traceback.format_exc(),sep="")
    return "Exception:\n" + except_message + "Stack Frame:\n" + stack_message


def killpyc(arg,dirname,fnames):
    for fname in fnames:
        ext = os.path.splitext(fname)[1].lower()
        if ext in [ ".pyo", ".pyc"]:
            doomed_path = os.path.join(dirname,fname)
            try:
                os.unlink(doomed_path)
                sys.stdout.write("Deleted %s\n"%doomed_path)
                sys.stdout.flush()
            except (KeyboardInterrupt,SystemExit):
                #If a Ctrl-c or System exit is triggered go ahead
                #and exit
                raise
	    except:
                sys.stderr.write("Error trying to delete %s\n"%doomed_path)
                sys.stderr.write("Excuse is %s:\n"%excuse())
                sys.stderr.flush() 

os.path.walk(".",killpyc,tuple())
